lib.go

```diff
@@ -553,5 +558,16 @@ func DeserializeResponse(gasLimit uint64, deserCost types.UFraction, gasReport *
 		return err
 	}
 
+	// All responses that have sub-messages need their payload size to be checked
+	const ReplyPayloadMaxBytes = 128 * 1024 // 128 KiB
+	if response, ok := response.(hasSubMessages); ok {
+		for _, m := range response.SubMessages() {
+			// each payload needs to be below maximum size
+			if len(m.Payload) > ReplyPayloadMaxBytes {
+				return fmt.Errorf("reply payload larger than %d bytes: %d bytes", ReplyPayloadMaxBytes, len(m.Payload))
```

Comment:
Maybe something along the lines of "reply contains submessage with payload larger than [...]" and add the index of the submessage? Would be good to clarify the payload is not a field of the reply